File: OneTimePasswordLegacyTests/OTPTokenTests.m
//
//  OTPTokenTests.m
//  OneTimePassword
//
//  Created by Matt Rubin on 2/3/15.
//  Copyright (c) 2015 Matt Rubin. All rights reserved.
//

@import XCTest;
@import OneTimePasswordLegacy;


@interface OTPTokenTests : XCTestCase
@end


@implementation OTPTokenTests

- (void)testDefaults
{
    XCTAssertEqual([OTPToken defaultAlgorithm], OTPAlgorithmSHA1);
    XCTAssertEqual([OTPToken defaultDigits], 6U);
    XCTAssertEqual([OTPToken defaultInitialCounter], 0UL);
    XCTAssertEqual([OTPToken defaultPeriod], 30);
}

- (void)testInit
{
    OTPToken *token = [[OTPToken alloc] init];

    XCTAssertEqualObjects(token.name, @"");
    XCTAssertEqualObjects(token.issuer, @"");
    XCTAssertEqual(token.type, OTPTokenTypeTimer);
    XCTAssertEqualObjects(token.secret, [NSData new]);
    XCTAssertEqual(token.algorithm, OTPAlgorithmSHA1);
    XCTAssertEqual(token.digits, 6U);
    XCTAssertEqual(token.period, 30);
    XCTAssertEqual(token.counter, 0UL);
}

- (void)testConvenienceInit
{
    OTPTokenType type = OTPTokenTypeCounter;
    NSData *secret = [@"!" dataUsingEncoding:NSUTF8StringEncoding];
    NSString *name = @"Name";
    NSString *issuer = @"Issuer";

    OTPToken *token = [[OTPToken alloc] initWithType:type secret:secret name:name issuer:issuer];

    XCTAssertEqualObjects(token.name, name);
    XCTAssertEqualObjects(token.issuer, issuer);
    XCTAssertEqual(token.type, type);
    XCTAssertEqualObjects(token.secret, secret);
    XCTAssertEqual(token.algorithm, OTPAlgorithmSHA1);
    XCTAssertEqual(token.digits, 6U);
    XCTAssertEqual(token.period, 30);
    XCTAssertEqual(token.counter, 0UL);
}

@end

File: OneTimePasswordLegacyTests/OTPTokenTests.swift
//
//  OTPTokenTests.swift
//  OneTimePassword
//
//  Copyright (c) 2015 Matt Rubin and the OneTimePassword authors
//
//  Permission is hereby granted, free of charge, to any person obtaining a copy
//  of this software and associated documentation files (the "Software"), to deal
//  in the Software without restriction, including without limitation the rights
//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
//  copies of the Software, and to permit persons to whom the Software is
//  furnished to do so, subject to the following conditions:
//
//  The above copyright notice and this permission notice shall be included in all
//  copies or substantial portions of the Software.
//
//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
//  SOFTWARE.
//

import XCTest

class OTPTokenTests: XCTestCase {
    func testInit() {
        let token = OTPToken()

        XCTAssertEqual(token.name, "")
        XCTAssertEqual(token.issuer, "")
        XCTAssertEqual(token.type, OTPTokenType.timer)
        XCTAssertEqual(token.secret, Data())
        XCTAssertEqual(token.algorithm, OTPAlgorithm.sha1)
        XCTAssertEqual(token.digits, 6)
        XCTAssertEqual(token.period, 30)
        XCTAssertEqual(token.counter, 0)
    }
}
